pkg/workflow/templates.go

```diff
@@ -230,48 +142,64 @@ func generateSubchartAndAppDAGTasks(appGroup *v1alpha1.ApplicationGroup, namespa
 		return nil, err
 	}
 
-	task := v1alpha13.DAGTask{
-		Name:     utils.ConvertToDNS1123(app.Name),
+	task := *appDAGTaskHelper(app.Name, getTimeout(app.Spec.Release.Timeout), &hr)
+	task.Dependencies = func() (out []string) {
+		for _, t := range tasks {
+			out = append(out, utils.ConvertToDNS1123(t.Name))
+		}
+		return out
+	}()
+	tasks = append(tasks, task)
+
+	return tasks, nil
+}
+
+func appDAGTaskHelper(name string, timeout *v1alpha13.AnyString, hr *fluxhelmv2beta1.HelmRelease) *v1alpha13.DAGTask {
+	task := &v1alpha13.DAGTask{
+		Name:     utils.ConvertToDNS1123(name),
 		Template: HelmReleaseExecutorName,
 		Arguments: v1alpha13.Arguments{
 			Parameters: []v1alpha13.Parameter{
 				{
 					Name:  HelmReleaseArg,
-					Value: utils.ToAnyStringPtr(base64.StdEncoding.EncodeToString([]byte(utils.HrToYaml(hr)))),
+					Value: utils.ToAnyStringPtr(base64.StdEncoding.EncodeToString([]byte(utils.HrToYaml(*hr)))),
 				},
 				{
 					Name:  TimeoutArg,
-					Value: getTimeout(app.Spec.Release.Timeout),
+					Value: timeout,
 				},
 			},
 		},
-		Dependencies: func() (out []string) {
-			for _, t := range tasks {
-				out = append(out, utils.ConvertToDNS1123(t.Name))
-			}
-			return out
-		}(),
 	}
-	tasks = append(tasks, task)
-
-	return tasks, nil
+	return task
 }
 
 func generateSubchartHelmRelease(namespace string, a v1alpha1.Application, subchartName, version string) (*fluxhelmv2beta1.HelmRelease, error) {
 	chName := utils.GetSubchartName(a.Spec.Chart.Name, subchartName)
+	hr := helmReleaseHelper(&a, namespace, chName, chName, subchartName, version)
+
+	val, err := subchartValues(subchartName, a.GetValues())
+	if err != nil {
+		return nil, err
+	}
+	hr.Spec.Values = val
+	return hr, nil
+}
+
+func helmReleaseHelper(app *v1alpha1.Application, namespace, objMetaName, chName, releaseName, version string) *fluxhelmv2beta1.HelmRelease {
```

Comment:
Instead of passing `app` we could just pass in `app.Spec.Release`

Reply:
maybe rename the functions to `<xyz>Builder` instead of `<xyz>Helper`

Reply:
Fixed